### PR TITLE
Bug/gh214 no update update on disable

### PR DESCRIPTION
**Pull request by blackboxsw**
WIP branch for review

Fixes: #214   Don't apt update on disable
Fixes: #234 fips: actually install dependent packages

### CODE REVIEW

File: uaclient/entitlements/fips.py [hunk 1]
```diff
@@ -52,24 +52,22 @@ def enable(self):
                 name=self.name, series=series)
             apt.add_ppa_pinning(
                 repo_pref_file, repo_url, self.repo_pin_priority)
-        return True
```
Comment (blackboxsw) [inline uaclient/entitlements/fips.py]: We weren't installing FIPS packages due to early return

File: uaclient/entitlements/tests/test_fips.py [hunk 2]
```diff
@@ -35,49 +38,78 @@
...
-        cfg.write_cache('machine-token', FIPS_MACHINE_TOKEN)
-        cfg.write_cache('machine-access-fips', FIPS_RESOURCE_ENTITLED)
-        entitlement = FIPSEntitlement(cfg)
-        # Unset static affordance container check
-        entitlement.static_affordances = ()
-
-        with mock.patch('uaclient.apt.add_auth_apt_repo') as m_add_apt:
+    @mock.patch('uaclient.util.get_platform_info', return_value='xenial')
+    def test_enable_configures_apt_sources_and_auth_files_and_installs(
+            self, _platform_info):
```
Comment (OddBloke) [inline uaclient/entitlements/tests/test_fips.py]: If we did `s/return_value='xenial'/Mock(\0)/` in the decorator, we wouldn't get the mock passed in at all.

File: uaclient/entitlements/tests/test_fips.py [hunk 3]
```diff
@@ -35,49 +38,78 @@
...
+                        self.assertTrue(self.entitlement.enable())
 
-        add_apt_calls = [
+        add_apt_src_calls = [
             mock.call('/etc/apt/sources.list.d/ubuntu-fips-xenial.list',
                       'http://FIPS', 'TOKEN', None, 'APTKEY')]
         apt_pinning_calls = [
             mock.call('/etc/apt/preferences.d/ubuntu-fips-xenial',
                       'http://FIPS', 1001)]
+        install_packages = self.entitlement.packages
```
Comment (OddBloke) [inline uaclient/entitlements/tests/test_fips.py]: This doesn't need to be a copy as much as it does in the actual code above, but if it isn't then we're less likely to catch unexpected mutations like those I mentioned above.

**Comment by blackboxsw** [comment]
Will take review comments here and break this branch up into separate PRs.

**Comment by OddBloke** [comment]
(As @blackboxsw is working on this ATM, I'm going to close the PR so it doesn't show in the list as needing review.)

File: uaclient/entitlements/repo.py [hunk 4]
```diff
@@ -58,22 +58,23 @@ def enable(self):
...
-                print(
-                    'Installing {title} packages ...'.format(title=self.title))
-                util.subp(['apt-get', 'install'] + self.packages, capture=True)
-        except util.ProcessExecutionError:
-            self.disable(silent=True, force=True)
-            logging.error(
-                status.MESSAGE_ENABLED_FAILED_TMPL.format(title=self.title))
-            return False
+            install_packages.append('ca-certificates')
+        if install_packages:
```
[the remark this answers is not in the document]
Comment (blackboxsw) [reply uaclient/entitlements/repo.py]: Good point. Relying on repo.enable instead. dropping fips.enable

File: uaclient/entitlements/repo.py [hunk 5]
```diff
@@ -58,22 +58,23 @@ def enable(self):
                 name=self.name, series=series)
             apt.add_ppa_pinning(
                 repo_pref_file, repo_url, self.repo_pin_priority)
+        install_packages = self.packages
         if not os.path.exists(apt.APT_METHOD_HTTPS_FILE):
-            util.subp(['apt-get', 'install', 'apt-transport-https'],
-                      capture=True)
+            install_packages.append('apt-transport-https')
```
[the remark this answers is not in the document]
Comment (blackboxsw) [reply uaclient/entitlements/repo.py]: Dan, you are correct here. I found this out in later testing on this dead branch. We do need two phases and I'll try to make them depend on the version of apt.

File: uaclient/entitlements/tests/test_fips.py (hunk 3 above)
Comment (blackboxsw) [reply uaclient/entitlements/tests/test_fips.py]: +1 will resolve to separate this in a new branch that I will link here.

Comment (blackboxsw) [reply uaclient/entitlements/tests/test_fips.py]: Resolved these comments in https://github.com/CanonicalLtd/ubuntu-advantage-client/pull/263

**Pull request closed without merging**